### ISSUE TITLE
masking the padded frames

### ISSUE THREAD

**Issue by AzamRabiee** [keithito/tacotron#31]
@keithito I've your original code on Blizzard2012! Nothing is changed! It seem that some thing is wrong with the silence at the end. 
You may tack a look at the waves at different global steps.  
![blizzard_generated_maxiter_100_eval-x-0](https://user-images.githubusercontent.com/29938304/29649758-34b69e88-88d1-11e7-9502-cc98bf6c8219.png)
or listen to them:
[Synthesized.zip](https://github.com/keithito/tacotron/files/1247734/Synthesized.zip)

**Comment by keithito** [comment]
Yes, that is very strange. Can you attach the alignments for step 151K and 615K? I don't see how it could have a good alignment and still generate audio with so much silence at the start and garbage at the end. It could be that something went wrong in this particular training run. You may want to try restarting training at a checkpoint where things were still good (using the --restore_step flag).

I'm not sure that masking the padding frames will help. It might make things worse -- right now the model learns to predict the padding frames (which generate silence) at the end of the audio. Without them, it might just generate garbage at the end. But please feel free to give it a try and report back on whether it works! I think you it would just be a small change here:
https://github.com/keithito/tacotron/blob/master/models/helpers.py#L43

Another option would be to train a classifier to predict when to stop generating audio. Right now at eval time, it's looking for an all-zero frame to determine when to stop. But since everything is real-valued, that never happens.
https://github.com/keithito/tacotron/blob/master/models/helpers.py#L27

**Comment by AzamRabiee** [comment]
Thanks for your complete reply! 
I will try your suggestions and let you know results, soon.

But for now, please just take a look at the alignments for the short sentence at 67k, 109k, 151k, and 615k gs:
![eval-0-align](https://user-images.githubusercontent.com/29938304/29737551-cf5a4e38-8a4a-11e7-9f83-d56ed1118e51.png)
corresponding waves
[Waves_ShortSentence.zip](https://github.com/keithito/tacotron/files/1253334/Waves_ShortSentence.zip)


And,
Compare it with long sentence at same global steps:
![eval-9-align](https://user-images.githubusercontent.com/29938304/29737562-f13bd63e-8a4a-11e7-8dc2-535906daa847.png)
corresponding waves
[Waves_LongSentence.zip](https://github.com/keithito/tacotron/files/1253335/Waves_LongSentence.zip)

Please let me know if you have any idea.

**Comment by AzamRabiee** [comment]
@keithito as you said, it seems that something went wrong in "this particular" training run.
cause I did not faced with this issue in my non-English Tacotron!

**Comment by qclu** [comment]
@keithito 
Firstly, thank you for your nice share!
For now, I am also facing with the problem that audio synthesized is fixed length and appended with  silence at the end. And if I try with long sentences, at the end, it seems the pronunciation is squeezed together.
You said above---'It might make things worse -- right now the model learns to predict the padding frames (which generate silence) at the end of the audio.' If so, does it means that if I keep the train running, the problem could be solved? The steps is about 300k now, and the audio file sounds good except for the problem mentioned above.

**Comment by keithito** [comment]
No, training longer will probably not fix the problem. However, increasing the max_iters hyperparameter at eval time will likely help with the issue where things are squeezed together in long sentences.

**Comment by qclu** [comment]
@keithito 
I have tried increased the max_iters hyperparameter at eval time. It can generate longer wave, but still with fixed length. I wonder if I can get audio with length varied according length of text. Or, I can only resize audio file after synthesized to remove silience at the end.
